iis-common/src/test/java/eu/dnetlib/iis/common/java/io/SequenceFileTextValueReaderTest.java

```diff
@@ -0,0 +1,80 @@
+package eu.dnetlib.iis.common.java.io;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.isA;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+import org.apache.hadoop.fs.Path;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+
+import eu.dnetlib.iis.common.IntegrationTest;
+
+@Category(IntegrationTest.class)
+public class SequenceFileTextValueReaderTest {
+
+    private static final String SEQUENCE_FILES = "sequence_files";
+    private static final String ONE_FILE = SEQUENCE_FILES+"/part-00000";
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    //------------------------ TESTS --------------------------
+
+    @Test
+    public void readFile() throws Exception {
+        // given
+        SequenceFileTextValueReader reader = newReader(ONE_FILE);
+
+        // execute
+        ArrayList<String> items = readAll(reader);
+
+        // assert
+        assertThat(items, contains("a", "bb", "ccc"));
+    }
+
+    @Test
+    public void readDirectory() throws Exception {
+        // given
+        SequenceFileTextValueReader reader = newReader(SEQUENCE_FILES);
+
+        // execute
+        ArrayList<String> items = readAll(reader);
+
+        // assert
+        assertThat(items, contains("a", "bb", "ccc", "a", "bb", "ccc"));
+    }
+
+    @Test
+    public void closeReader() throws Exception {
+        // given
+        SequenceFileTextValueReader reader = newReader(ONE_FILE);
+
+        // execute
+        reader.close();
+
+        // assert
+        thrown.expectCause(isA(IOException.class));
+        reader.hasNext();
```

Comment:
It's more complicated than that. Maybe this test is not very useful?
`hasNext` doesn't have to throw after closing the iterator - in particular it will not throw if we are at the end of the iteration already, it will just return `false`, and that's OK.
What we are really interested in is that the `close` on reader does really close the underlying resources.
The exception from `hasNext` is only proxy for that. So in my mind it is really **given** a reader, **when** we close the reader, **assert** that to the best of our knowledge the underlying iterator was also closed.